### PR TITLE
Pubish Subscribe support for S7

### PR DESCRIPTION
**Pull request by glcj**
Hi,

I have tested the connection with an S7-300 (314) + CP 343-1, and I have not been able to make the connection.

When checking with WireShark I found that the dst-tsap and the src-tsap were inverted.

Cheers

### CODE REVIEW

**Comment by chrisdutz** [comment]
Hi César, thanks for this great PR ... we'll evaluate this.

File: vim.exe.stackdump [hunk 1]
```diff
@@ -0,0 +1,19 @@
+Stack trace:
```
Comment (sruehl) [inline vim.exe.stackdump]: @glcj a apache header is missing for this file ;)
https://www.apache.org/legal/src-headers.html

Comment (glcj) [reply vim.exe.stackdump]: Ready for the next commit, thanks.

**Comment by etiennerobinet** [comment]
Will this branch be merged? Or is the /develop branch supporting S7 Subscription?

**Comment by chrisdutz** [comment]
No this branch can't be merged as the files it refers to have been deleted ... It's a work in progress which will be converted to mspec and the new generated drivers as soon as it's done.

**Comment by chrisdutz** [comment]
Any progress on bringing this to mspec?

**Comment by glcj** [comment]
> 
> 
> Any progress on bringing this to mspec?

Hi Chris,

At this moment I am migrating Merlot to use the PLC4X driver model, as soon as I finish I will carry out load tests for the S7 driver, to later migrate to mspec and be in line with the rest of the project.

Work in progress... :-)

**Comment by mahendra-korat** [comment]
Any progress on this? I need to start trials with my PLCs.

**Comment by chrisdutz** [comment]
I guess Cesar is currently consumed with other work topics ... if this is a high priority requirement for you and your company you could consider contracting someone of the community to work on this (I for my part don't have the necessary hardware ... my S7-1200 doesn't support this form of communication)

**Comment by glcj** [comment]
Hello,

The features noted are functional in version 0.6.1 (actually the "s7alarm" branch).

The software is tested for S7-300 & S7-400. 

S7-1500 and S7-1200 stations use or do not have a notification system.

By next week I must finish the tests on an S7-400H system, which would transform this version of the driver into the most complete one outside the Siemens ecosystem.

As Chris points out, the next step is to go to the mspec, which is an interesting job, but needs to be funded as you point out.

I will be happy to help you with testing, documenting everything on the PLC4X mailing list.

Best regards,

**Comment by sruehl** [comment]
superseeded by https://github.com/apache/plc4x/pull/287

**Pull request closed without merging**